yrs/src/store.rs

```diff
@@ -254,6 +258,46 @@ impl Store {
             None
         }
     }
+
+    /// Consumes current block slice view, materializing it into actual memory layout,
+    /// splitting underlying block along [BlockSlice::start]/[BlockSlice::end] offsets.
+    ///
+    /// Returns a block created this way, that represents the boundaries that current [BlockSlice]
+    /// was representing.
+    pub(crate) fn materialize(&mut self, mut slice: BlockSlice) -> BlockPtr {
```

Comment:
This method takes a block slice and slices the underlying block accordingly to a ranges defined by that slice.